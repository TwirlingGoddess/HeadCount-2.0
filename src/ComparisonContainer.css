
.comp-container {
  background-color: #C2B2B4;
  display: flex;
  justify-content: space-around;
  flex-wrap: wrap;

}

.comp-card {
  border: 2px solid chartreuse;
  width: 15vw;
  background-color: #ffffff;
}