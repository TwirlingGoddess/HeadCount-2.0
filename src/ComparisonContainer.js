import React from 'react';

const ComparisonContainer = () => {
  return (
    <div> ComparisonContainer </div>
  )
};

export default ComparisonContainer;