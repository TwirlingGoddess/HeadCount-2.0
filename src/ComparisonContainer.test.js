import ComparisonContainer from './ComparisonContainer.js';
import React from 'react';
import { shallow } from 'enzyme';

describe ('Comparison Container tests', () => {
  it('matches the snapshot when no cards have been clicked', () => {
    const mockArray = []
    const wrapper = shallow(<ComparisonContainer cardArray={mockArray} />);
    expect(wrapper).toMatchSnapshot()
  })

  it('displays a card in the comparison container when a district card is clicked on', () => {
    const mockDistrict = {
      'ACADEMY 20':
        { 2004: 0.302, 2005: 0.267, 2006: 0.354, 2007: 0.392, 2008: 0.385, 2009: 0.39, 2010: 0.436, 2011: 0.489, 2012: 0.479, 2013: 0.488, 2014: 0.49 }
    }
    const mockArray = [mockDistrict]
    
    const wrapper = shallow(<ComparisonContainer cardArray={mockArray}/>);
    const expected = wrapper.find('.comp-card').length
    expect(expected).toBe(1)
  });

  it('displays 2 district cards and 1 comparison card in the comparison container when two district cards  have been clicked on', () => {
    const mockDistrict1 = {
      'ACADEMY 20':
        { 2004: 0.302, 2005: 0.267, 2006: 0.354, 2007: 0.392, 2008: 0.385, 2009: 0.39, 2010: 0.436, 2011: 0.489, 2012: 0.479, 2013: 0.488, 2014: 0.49 }
    }
    const mockDistrict2 = {
      'Colorado':
        { 2004: 0.303, 2005: 0.267, 2006: 0.354, 2007: 0.392, 2008: 0.385, 2009: 0.39, 2010: 0.436, 2011: 0.489, 2012: 0.479, 2013: 0.488, 2014: 0.45 }
    }
    const mockArray = [mockDistrict1, mockDistrict2]

    const wrapper = shallow(<ComparisonContainer cardArray={mockArray} />);
    const expectedDistrictCards = wrapper.find('.comp-card').length
    const expectedTotalCards = wrapper.find('.total-card').length
    expect(expectedDistrictCards).toBe(2)
    expect(expectedTotalCards).toBe(1)
  })

})

