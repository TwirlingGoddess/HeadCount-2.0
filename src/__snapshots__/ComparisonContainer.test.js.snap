// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comparison Container tests matches the snapshot when no cards have been clicked 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ComparisonContainer
    cardsToCompare={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
